### ISSUE TITLE
How to see which species groups have been counted in the sample download of 15 min counts?

### ISSUE THREAD

**Issue by chrisvanswaay** [BiologicalRecordsCentre/ABLE#481]
**Warehouse**:
- [x] Create a new sample warehouse attribute - an array of species group ID numbers. 

**App**:
- [x] Attach selected species group information to records. 
   - [x] Add a species group filter dialogue to the species search page header. 
      - [x] Copy over the previous survey settings to new surveys (save the selected options to the survey metadata but also update global app store settings for subsequent surveys). 
      - [x] Show how many species groups are enabled as filter label. e.g. "<btn>Groups (1)</btn>"
      - [x] Don't allow unchecking a species group filter if a species of this group is already in the survey list.
   - [x] On survey Finish, show a popup with all the species groups. 
      - [x] Don't allow unchecking a species group if a species of this group is already in the survey list. 
      - [x] Don't show the popup if all the selected species groups are present in the survey list. 
      - [x] Title "Have you counted these species groups?". 
      - [x] Save this setting to the survey attribute for upload.
      - [x] On upload, send the counted species groups list based on the groups' finish popup selection AND what is currently in the species list (if new species were later added after the popup).
      - [x] Once the Finish is pressed and the survey is completed, then the filter options for that survey are locked so no further species groups can be added.
   - [x] Remove App-level favourite species group setting - add a temporary placeholder to inform this was moved to per-survey and can be found in the species search page header.

**Website**:
- [ ] BiologicalRecordsCentre/ABLE#552

When I make a download of the 15-minute counts, then how can I see which species groups were counted? I can't use the occurrences for that, because you can do a count for butterflies and see nothing (so no record in the occurrences). There is a field called taxa, but I get the impression this is the number of taxa counted. So I need somehow to be able to see for which species groups a count has been made. Of course some people can do a count for more species groups, e.g. butterflies and dragonflies.

**Comment by chrisvanswaay** [comment]
Anyone?

**Comment by JimBacon** [comment]
I have no knowledge of this survey, at present, so am not the best person to try to answer the question. If it was not designed to record absence then this may be difficult.

**Comment by chrisvanswaay** [comment]
Thanks @JimBacon .  I can download samples and occurrences, I guess @johnvanbreda prepared that query. From the species in the occurrences-download I can see that it's not only butterflies (but also dragonflies etc). Before starting the count a recorder has to indicate which speciesgroups will be monitored. I need to know which species groups were monitored in a sample. @johnvanbreda  am I right that you made those queries? Is it possible to adapt them in a way that I can see which speciesgroup was counted in a sample?

**Comment by johnvanbreda** [comment]
@kazlauskis presumably the question about species groups is collected by the app - do you know if this information gets stored in the database anywhere?

**Comment by kazlauskis** [comment]
@chrisvanswaay @johnvanbreda the app allows users to filter the species lists to selected species groups but that is about it. This filter is part of the app setting and isn't uploaded to the database as part of the records. I don't see why we couldn't attach the species groups' information to the surveys, though.

**Comment by chrisvanswaay** [comment]
@kazlauskis Thanks. Hope this can be implemented, it allows the distinction of zero-counts and these are important.

**Comment by kazlauskis** [comment]
Sure, we can add the species groups to the surveys.

**Comment by DavidRoy** [comment]
@chrisvanswaay @kazlauskis I'm not sure that is sufficient.  The app species list settings do not necessarily confirm the groups recorded during a count.  I think we should have this as a setting in the 'Additional Details' part of the recording forms.  Also, we should indicate on the species search page which groups are being searched (there is room for this?).

**Comment by chrisvanswaay** [comment]
It might be the question at the end of the survey (that's the way they do it in Avimap), in which you can also distinguish between complete surveys for a species group, and incomplete surveys for a species group. At the end you usually know which speciesgroups you did count and if complete or not. The result of that question could be saved.

**Comment by DavidRoy** [comment]
I agree that would work well and make sure we capture this information
@kazlauskis can you implement in the next version

**Comment by kazlauskis** [comment]
Do we also want to ask for 'completeness' or just the species groups?

**Comment by DavidRoy** [comment]
Just a list of species groups, same as for the app setting which subdivides (moths and day-flying moths).

Title = Please select the species groups that were included in your survey count.  This is important to distinguish zero counts from 'not included'

The page can default to the app setting

I'm not sure that asking about 'incomplete lists' is helpful.  We there assume that all butterflies, moths/day-flying moths, dragonflies, bumblebees were counted if seen/detected.

**Comment by kazlauskis** [comment]
OK 👍   

I am thinking about how to improve this for the majority of the users who will probably record only one species group. If the user has selected only one group, then the app won't allow recording any other groups, so maybe we show this question to only those who have chosen multiple groups? In that case, we would still attach the only species group to the record in the background.

**Comment by chrisvanswaay** [comment]
That's a good idea.

**Comment by chrisvanswaay** [comment]
@kazlauskis @DavidRoy 
Has this issue been solved? It is one of the most important features for the analysis of 15 min counts: we have to know when nothing of a species group has been seen (a zero count). Please let me know if, and if yes, when this will be fixed. I don't think it has to be complicated to do, and I don't really mind how you do it, as long as we can distinguish zero counts per species group later in the downloads.

**Comment by kazlauskis** [comment]
No, we haven't worked on it yet but we should be able to do it in April. Adding it to my calendar now!

**Comment by chrisvanswaay** [comment]
@kazlauskis Flagging this up, as it's one of the most important features for future analysis of the data. Would be a shame to miss this for coming season.

**Comment by kazlauskis** [comment]
OK, we'll be working on the app next month and should be able to do this.

**Comment by kazlauskis** [comment]
We've updated the ticket description. It explains the thinking behind the implementation so please check if that also makes sense to you all. 

In the end, on survey finish users will be presented with a list of species groups that they have the filter turned on but haven't recorded any. From the UX perspective, most people won't notice a difference and will also have a more convenient place to change their favourite species groups.

**Comment by DavidRoy** [comment]
@chrisvanswaay can you review the edits at the top of the issue?

**Comment by chrisvanswaay** [comment]
If you have seen nothing at all, then the question "Have you also counted these species groups?" might feel a bit strange. Maybe remove the word 'also'?

**Comment by Vilius-Stankaitis** [comment]
@chrisvanswaay  Hi, I have finished building this feature. Could you test it? 

https://butterflies.app.flumens.io/demo.html

**Comment by chrisvanswaay** [comment]
@Vilius-Stankaitis Thanks. I tried, but not sure what I should see. I noticed I cannot add species groups in the settings (it says 'This option was moved to the species search page.', but not sure where that is, to be honest). I entered one butterfly, and could also upload it without any extra questions. What am I missing?
And also important: how will this appear in the sample download?

**Comment by CrisSevilleja** [comment]
I tried the same as @chrisvanswaay and I couldn't find that question if other taxa groups were recorded. Also, I noticed the other groups disappeared from the selection menu, maybe it has to be for selecting the taxa to be recorded before starting the count? I think we are missing something. Thanks @Vilius-Stankaitis for going further with this.

**Comment by DavidRoy** [comment]
@Vilius-Stankaitis I found the same.  I think what's missing is:

On survey Finish, show a popup with all the species groups.
 Don't allow unchecking a species group if a species of this group is already in the survey list.
 Don't show the popup if all the selected species groups are present in the survey list.
 Title "Have you counted these species groups?".
 Save this setting to the survey attribute for upload.
 On upload, send the counted species groups list based on the groups' finish popup selection AND what is currently in the species list (if new species were later added after the popup).
 Once the Finish is pressed and the survey is completed, then the filter options for that survey are locked so no further species groups can be added.

**Comment by DavidRoy** [comment]
I also think we should retain the App-level favourite species group setting as this has wider use, e.g. for transect recording and/or the guide (in future).

**Comment by CrisSevilleja** [comment]
so basically you need to decide which species group you want to record in advance and not possible to add them when you started a count. I think it is logical but we should mention it somewhere, in a text box informing about this on the app. 

The current text on the species group: 'This option was moved to the species search page.' 
is a bit vague and generates confusion. I would change it to something like "Species group(s) already selected for the survey."

**Comment by chrisvanswaay** [comment]
I think it still should be possible to give the species groups in the settings (as it used to be), as you will choose the species from that list. Once finished the question should be if you counted all species of that species group (for the other ones you won't be able to enter a record, in theory, as the species will not be in your list). But it is possible that I want two species groups (say butterflies and dragonflies), but at the end only entered all data for butterflies, and for dragonflies only one species I am able to recognise. Then I did not do a full count for dragonflies.

**Comment by DavidRoy** [comment]
@Vilius-Stankaitis - let me know if you want a quick meeting to discuss the requirement for this

**Comment by Vilius-Stankaitis** [comment]
Hi all, thank you for the feedback. 

There is a quick demo of how it works. The intention was to make it smart and not show an alert every time.

https://zube.io/files/biologicalrecordscentre/451f757d3603b086e9beee4826c2cc82-screen-recording-2023-04-24-at-16-31-12.mov

**Comment by DavidRoy** [comment]
That makes sense.  When I first tested I didn't get the pop-up when selecting 'finish'

**Comment by chrisvanswaay** [comment]
Ah, ok, the problem with my testing is that a count refuses to finish as I have no area selected (and I don't get the tools to do this when I click on the map). I guess this has to do with the demo, and is the reason I never saw the popup.
A point is that it is not clear to most that the three lines in the right upper corner refer to the species groups. Would there be another, more intuitive solution?

**Comment by Vilius-Stankaitis** [comment]
@DavidRoy Yes, it's because the default setting for the species groups is 'Butterflies'.  I could pre-set all species groups as default with some explanatory text and show it the first time.

@chrisvanswaay you could uncheck gps toggle from the map and manually draw area. 

Initially idea was to add "Groups" label instead of icon, but we decided you save space and keep it simple.

![Screenshot 2023-04-24 at 17.00.12.png](https://zube.io/files/biologicalrecordscentre/b13784d9bf552c788d2bf8e41c8cc9d7-screenshot-2023-04-24-at-17-00-12.png)![Screenshot 2023-04-24 at 17.00.28.png](https://zube.io/files/biologicalrecordscentre/7e1b97bb0340a3871e830f28df71246d-screenshot-2023-04-24-at-17-00-28.png)

**Comment by Vilius-Stankaitis** [comment]
@chrisvanswaay In the sample download it appears as a list of species groups eg. ("Butterflies", "Moths", "Dragonflies")

**Comment by Vilius-Stankaitis** [comment]
I have updated the demo app. Now it will print species group when the upload button is tapped. Upload to warehouse functionality is temporally disabled. Also, map validation is disabled.

Reset the app before testing.

**Comment by chrisvanswaay** [comment]
I have tried and as far as I can see it works as expected. Thanks! I guess the download is not adapted yet?

**Comment by Vilius-Stankaitis** [comment]
No, there is another ticket (#724) for this task. Maybe @DavidRoy could assign someone for implementing this.

**Comment by Vilius-Stankaitis** [comment]
Also, should I change the "three lines" icon to "Groups"? This will look stuffy with other languages
![Screenshot 2023-04-25 at 11.37.14.png](https://zube.io/files/biologicalrecordscentre/8d1dd107197040b5d38429d5fd069422-screenshot-2023-04-25-at-11-37-14.png) 

I have also added an additional explanation image to the Settings page. Would this be more clear?
![Screenshot 2023-04-25 at 11.30.19.png](https://zube.io/files/biologicalrecordscentre/5fc8313ed4f1adc3a053770fab40f0f3-screenshot-2023-04-25-at-11-30-19.png)

**Comment by chrisvanswaay** [comment]
Yes, that's better. One thing: if I understand it well at the moment it means that if you have entered one species of a speciesgroup, you are considered to have counted all species. It is not possible to count one species of a species group (as me: I  can do all butterflies, but only recognize a few dragonflies, meaning I cannot count them). You could also image that the last question in the popup box would be: "Did you count all species of these species groups". @DavidRoy what do you think?

**Comment by DavidRoy** [comment]
@chrisvanswaay I think that's a new requirement but could get quite complicated.  We have two options:
1. release the current update so it is available
2. wait until we can decide on extending the functionality.  I suggest that we would need an interface for users to choose an options for each taxon group, e.g. 'All species counted' or 'A sub-set of species counted'

What do you think?

**Comment by chrisvanswaay** [comment]
No doubt, releasing this is most important. Lets keep it in mind for later.
So 1. release the current update so it is available

**Comment by Vilius-Stankaitis** [comment]
Ok

**Comment by CrisSevilleja** [comment]
@Vilius-Stankaitis replying to this previous comment:

> Also, should I change the "three lines" icon to "Groups"? This will look stuffy with other languages ![Screenshot 2023-04-25 at 11.37.14.png](https://camo.githubusercontent.com/6c7df15b1e34c6c6cbb63684ea81ec97bcf63cb7aefa20c433c1183815424442/68747470733a2f2f7a7562652e696f2f66696c65732f62696f6c6f676963616c7265636f72647363656e7472652f38643164643130373139373034306235643338343239643566643036393432322d73637265656e73686f742d323032332d30342d32352d61742d31312d33372d31342e706e67)
> 
> I have also added an additional explanation image to the Settings page. Would this be more clear? ![Screenshot 2023-04-25 at 11.30.19.png](https://camo.githubusercontent.com/20618d3f07da062fa2bb56f6b2eb1c26da232fcaaa4b51bc7b38a7879ad6ebf7/68747470733a2f2f7a7562652e696f2f66696c65732f62696f6c6f676963616c7265636f72647363656e7472652f35666338333133656434663161646333613035333737306661623430663066332d73637265656e73686f742d323032332d30342d32352d61742d31312d33302d31392e706e67)

I reviewed this part I like it a lot thank you, know it is much better to understand with the explanation image. However, could you edit the description text when selecting Species Groups under App settings to: 
"This option was moved to the species search page inside the surveys. When a survey starts, you can select which species group you are recording." 
and below the explanatory image you added before. I think with that is perfectly understood and also clear for translation into other languages. 

About the stuffy appearance of other languages, maybe you can give a bit more space for that string. The majority of the languages uses 6-7 letters for groups. 

Thank you!

**Comment by chrisvanswaay** [comment]
@DavidRoy Made a download of the 15 minute counts, but how can I see in the samples which groups have been counted? 
Now the sample file looks like this:

 $ Visit Sample ID  : int  6131122 6132725 6134702 6134703 6134704 6134705 6134706 6134707 6134708 6134709 ...
 $ Date             : chr  "08/07/2019" "08/07/2019" "29/06/2019" "04/07/2019" ...
 $ Sample centroid  : chr  "SK42969285" "SU61678967" "SU61428919" "SU61728965" ...
 $ Geom             : chr  "LINESTRING(-1.354157 53.439502,-1.355389 53.421986)" "POLYGON((-1.111025 51.602769,-1.110628 51.602689,-1.110869 51.602313,-1.111245 51.602399,-1.111025 51.602769))" "LINESTRING(-1.11560663 51.59778638,-1.11549467 51.59796382,-1.11534308 51.59807273,-1.11533478 51.59826795,-1.1"| __truncated__ "LINESTRING(-1.11021104 51.60285579,-1.10981955 51.60257489,-1.11006507 51.60254546,-1.11006961 51.60233512,-1.1"| __truncated__ ...
 $ Recorded by      : chr  "" "Roy, David" "Roy, David" "Roy, David" ...
 $ Area sampled (m²): num  1928556 1256 2413 1421 1974 ...
 $ Start Time       : chr  "11:06" "21:41" "14:01" "12:34" ...
 $ End Time         : chr  "11:06" "21:42" "14:14" "12:42" ...
 $ Temp (Deg C)     : chr  "" "" "" "" ...
 $ % Cloud          : int  NA NA NA NA NA NA NA NA NA NA ...
 $ Wind Direction   : chr  "" "" "" "" ...
 $ Taxa             : int  0 1 0 0 1 2 1 1 1 0 ...
 $ Individuals      : int  0 1 0 0 2 2 1 1 1 0 ...

**Comment by DavidRoy** [comment]
@chrisvanswaay we haven't updated the download files yet.  I'll create a new issue for this

**Comment by chrisvanswaay** [comment]
thanks
